src/System.Runtime/ref/System.Runtime.cs

```diff
@@ -1968,6 +1984,7 @@ public partial class RankException : System.SystemException
         public bool TryCopyTo(System.Span<T> destination) { throw null; }
         public ref partial struct Enumerator
         {
+            private int _dummy;
```

Comment:
The Enumerator struct contains Span field. It should be treated same way as Span and have object field.

This applies to all structs with Span fields.

Reply:
I should probably update the tool to special case Span and ReadOnlySpan.

Reply:
Perhaps it is easier to hack the tool so that it thinks Span/ReadonlySpan actually contain an extra `private object _dummy` ?

Reply:
I'm going to teach the tool to recognize "ByReference<T>" as hold a reference.

Reply:
That could add an object into `TypedReference`. 
It is technically breaking. `TypedReference*` It is a very corner case and maybe fragile/broken anyways, so I am not sure how much we care.

Reply:
The rule https://github.com/dotnet/buildtools/pull/1859/commits/4b80362df58e1ef45e58ca482b718084baddbcf2 produced the expected output 
https://github.com/dotnet/corefx/pull/26264/commits/6ef954562c71fe7f3de4db7407d3d4ec58825dd7.

TypedReference didn't change because it only contains an IntPtr not a ByReference.